src/interfaces/capnp/proxy-codegen.cpp

```diff
@@ -0,0 +1,472 @@
+
+                    std::ostringstream field_flags;
+                    field_flags << (!field.param ? "FIELD_OUT" : field.result ? "FIELD_IN | FIELD_OUT" : "FIELD_IN");
+                    if (field.optional) field_flags << " | FIELD_OPTIONAL";
+                    if (field.requested) field_flags << " | FIELD_REQUESTED";
+                    if (BoxedType(field_type)) field_flags << " | FIELD_BOXED";
+
+                    add_accessor(field_name);
+
+                    for (int i = 0; i < field.args; ++i) {
+                        if (argc > 0) client_args << ",";
+                        client_args << "M" << method.getOrdinal() << "::Param<" << argc << "> " << field_name;
+                        if (field.args > 1) client_args << i;
+                        ++argc;
+                    }
+                    client_invoke << ", ";
+
+                    if (field.exception.size()) {
+                        client_invoke << "ClientException<" << field.exception << ", ";
+                    } else {
+                        client_invoke << "MakeClientParam<";
+                    }
+
+                    client_invoke << "Accessor<Field" << Cap(field_name) << ", " << field_flags.str() << ">>(";
+
+                    if (field.retval || field.args == 1) {
+                        client_invoke << field_name;
+                    } else {
+                        for (int i = 0; i < field.args; ++i) {
+                            if (i > 0) client_invoke << ", ";
+                            client_invoke << field_name << i;
+                        }
+                    }
+                    client_invoke << ")";
+
+                    if (field.exception.size()) {
+                        server_invoke_start << "Make<ServerExcept, " << field.exception;
+                    } else if (field.retval) {
+                        server_invoke_start << "Make<ServerRet";
+                    } else {
+                        server_invoke_start << "MakeServerField<" << field.args;
+                    }
+                    server_invoke_start << ", Accessor<Field" << Cap(field_name) << ", " << field_flags.str() << ">>(";
+                    server_invoke_end << ")";
+                }
+
+                client << "    using M" << method.getOrdinal() << " = ProxyClientMethodTraits<" << method_prefix
+                       << "Params>;\n";
+                client << "    typename M" << method.getOrdinal() << "::Result " << method_name << "("
+                       << client_args.str() << ")";
+                client << ";\n";
+                def << "ProxyClient<" << message_namespace << "::" << node_name << ">::M" << method.getOrdinal()
+                    << "::Result ProxyClient<" << message_namespace << "::" << node_name << ">::" << method_name << "("
+                    << client_args.str() << ") {\n";
+                if (has_result) {
```

Comment:
Always true?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/10102#discussion_r230426764

> Always true?

has_result is false for any method that doesn't have an output named result, which is true for any method that returns void.